libs/feature/map/src/lib/map-context/component/map-context.component.ts

```diff
@@ -25,19 +26,35 @@ export class MapContextComponent implements OnChanges {
   @Input() context: MapContextModel
   @Output() featureClicked = new EventEmitter<Feature<Geometry>[]>()
 
+  resizeObserver = new ResizeObserver(() => {
+    if (this.context.extent) {
+      this.map.updateSize()
+      this.context.view = this.utils.getViewFromExtent(
+        this.context.extent,
+        this.map
+      )
+    }
+    this.service.resetMapFromContext(this.map, this.context)
+    this.resizeObserver.unobserve(this.mapEl)
+  })
+
   map: Map
+  mapEl: HTMLElement
 
   constructor(
     private service: MapContextService,
     private featureInfo: FeatureInfoService,
-    private manager: MapManagerService
+    private manager: MapManagerService,
+    private utils: MapUtilsService
   ) {
     this.map = manager.map
   }
 
   ngOnChanges() {
     if (this.context) {
-      this.service.resetMapFromContext(this.map, this.context)
+      this.mapEl = this.map.getTargetElement()
+      if (!this.mapEl) return
+      this.resizeObserver.observe(this.mapEl)
```

Comment:
Could we observe the map element in the low level map component instead or not ?

Reply:
Which component do you mean here?

Reply:
ok, so we've moved it to the `map.component`. Thanks @fgravin !